### PR TITLE
fix(indexing): order persisted collections by code point, not host collation

### PR DESCRIPTION
**Pull request by mohanagy**
Implements #715.

#715 must remain open until the squash merge is followed by successful push-triggered
protected CI on `next`.

## What this changes

`String.prototype.localeCompare` asks ICU, and ICU answers according to the locale Node takes
from `LC_ALL`/`LANG` at startup. Two machines therefore wrote **different bytes** for the same
inputs. Nine ordering sites now go through `compareUnicodeCodePoints`, the repository's single
comparator owner, reused unchanged:

```text
src/pipeline/indexing-outcomes.ts    6   outcomes, the four summary buckets, spi_diagnostics
src/shared/git.ts                    1   dedupePaths -> graph_build_freshness.git.dirty_files
src/pipeline/spi/diff-overlay.ts     1   the overlay's from|to|kind key
src/infrastructure/proof-report.ts   1   compare-summary discovery order
```

Twenty-five display sites across six files keep locale collation and now say so in the file,
so the next reader can tell a deliberate display sort from one that was missed.

Measured at `next @ b892eafc`, en-US vs sv-SE, wall clocks pinned:

```text
indexing-manifest.json                 1a221607de726d…   e489bae64c4e92…   differed
indexing-manifest.json (capability)    3a08365652373a…   3c24109b31a372…   differed
indexing-manifest.share-safe.json      222ca99813d191…   51afc42f91d89a…   differed
graph.madar (dirty worktree)           da6596da9f8a69…   654bee47e97da9…   differed
SPI diff overlay                       4e2e0bfe0cca50…   847576af598198…   differed
```

## What the ordering change does and does not touch

The persisted collections are **ordering-only**: de-duplication and merging happen before
every sort, so no sort can change which value survives into an artifact.

They are not free of positional *consumers*, and that is declared rather than denied.
`doctor.ts:864` and `doctor.ts:934` each filter the incomplete outcomes and take
`slice(0, 20)`, so this changes which twenty paths `madar doctor` lists — bounded display
slices, presentation only, and host-independent where the old order was not. Everything else
that reads outcomes builds a `Set` (`watch.ts:304`), iterates all of them (`watch.ts:334`),
builds a `Map` keyed by path (`indexing-generation.ts:151`) or filters and counts
(`indexing-manifest.ts:217`).

Object key order is likewise not simply insertion order: `JSON.stringify` emits integer-index
keys first, ascending, so capabilities `"10"` and `"2"` serialize as `"2"`, `"10"` whatever
the comparator returns. That is a language rule rather than an ICU one, so the
host-independence guarantee holds, and `C1-integer-keys` asserts it directly.

## Final merge evidence

Independent verdict:
`GO-715-ORDERING`

Exact PR head:
`5f94435deffcda059a16f5ebb4feef4aa4f8d90b`

Exact PR tree:
`bb782d9f158b0deb88b1430ec62e82826732ca32`

Base:
`next @ b892eafc916c8dfaddff13e585c901ebee187afe`

Tested merge result:
`a8653acb4d1ad61e17e5a49a12c4121a9b7536fb`

Protected CI:
workflow `32944573281`, six of six OS/Node lanes successful

Controls:
`57 / 57`

Affected suites:
`244 / 244`

Mutation inventory:
- 18 total
- 17 caught
- 1 equivalent fixture-only survivor
- 0 uncaught production-ordering mutations
- 0 skipped mutations

Equivalent survivor:
`M9 / C2-literal-capability` modifies only a fixture. The parser returns the literal
capability strings it receives, and the equality is asserted by the control so it fails if
that behavior changes.

Behavioral ordering controls:
- all nine production ordering sites covered;
- seven through behavioral plus static controls;
- `extraction_strategy_buckets` and `fallback_reason_buckets` through the static policy audit
  because the documented searched domain produced no discriminating locale;
- `reason_buckets` behaviorally covered through the az-AZ discriminator.

Locale search:
116 supported locales from the documented generated candidate set.
No claim of exhaustive ICU-locale enumeration.

Timeout invariant:
- non-Windows: `2 × 6000 + 3000 <= 15000`
- Windows: `2 × 13500 + 3000 <= 30000`
- negative and boundary controls pass.

Performance:
`−31.09% p50`

Artifact size:
`316659 bytes` on both measured arms

Review threads:
`2 / 2 resolved`

Scope:
This PR owns serialization ordering only. Locale-dependent selection and graph semantic
identity remain under #716/#719.

What this does not claim:
`graph.madar` semantic identity is not yet host-independent because the projector's
first-wins case-folding collision remains under #716.

No release is authorized by this merge.

## Compatibility

No schema, no version bump, no public signature change. `parseIndexingManifest` still accepts
every manifest it accepts today, including non-ASCII capabilities and arbitrary diagnostic
ids — the accepted domain is not narrowed. Ordering bytes change: code-point order puts
`Zebra` before `_internal` and before every lowercase name, so any artifact over a mixed-case
corpus re-orders on its next write. Rollback is a plain revert with no migration in either
direction.

Refs #716, #719.

### CODE REVIEW

**Comment by mohanagy** [comment]
### Qualification complete — awaiting merge authorization

```text
head        645d6959f5eee9c1dac1761a7fbd28a14ae6df17
tree        a866cd0e4aaa4791c5669f7e1e2f6e5494c7af83
base        origin/next @ b892eafc916c8dfaddff13e585c901ebee187afe
FINAL       GO-715-ORDERING
```

Protected CI, every run bound to that exact head:

```text
validate (macos-latest,   Node 20)   success   98071881340
validate (macos-latest,   Node 22)   success   98071881354
validate (ubuntu-latest,  Node 20)   success   98071881410
validate (ubuntu-latest,  Node 22)   success   98071881241
validate (windows-latest, Node 20)   success   98071881353
validate (windows-latest, Node 22)   success   98071881413
CodeRabbit                           pass
```

The Windows lanes matter here: ICU on Windows follows the system locale and ignores
`LC_ALL`, so the cross-host arm's divergence assertion is skipped there by design and the
hermetic order assertion carries the guarantee. Both lanes are green.

Evidence: controls fail 35/45 at the base and pass 45/45 here; `tsc --noEmit` clean; 244
tests across the ten affected suites; 10 of 12 mutations caught, with both survivors
documented as uncatchable; artifact size identical at 316659 bytes on both locale arms;
performance −30.98% p50 (p10 −32.96, p90 −29.22) on a paired in-process A/B.

Not merging — that is the maintainer's gate. If the head moves, this authorization is void.

File: src/pipeline/export/overview-navigation.ts [hunk 1]
```diff
@@ -27,6 +27,13 @@ function overviewNodeHref(communityPagesDirname: string, communityId: number, no
   return `${communityPagesDirname}/community-${communityId}.html#${pageMode === 'summary' ? nodeAnchorId(nodeId) : encodeURIComponent(nodeId)}`
 }
 
+/**
+ * Display ordering: the overview navigation is HTML for a person to read, never
+ * parsed back by Madar, so collation is the right order and `localeCompare`
+ * stays here and in `buildOverviewCommunityLinks` below. Persisted collections
```
[the remark this answers is not in the document]
Comment (mohanagy) [reply src/pipeline/export/overview-navigation.ts]: **Corrected** in `ea4d636e`.

Verified first: `grep -rn buildOverviewCommunityLinks src/` returns nothing, so the name was invented — the file has `overviewNodeHref`, `buildOverviewTopNodes` and `buildOverviewSearchIndex`, and the two `localeCompare` sites are the top-node tie-break at :51 and the search-index sort at :67.

The comment now names those two sorts instead of a function that does not exist.

**Comment by mohanagy** [comment]
### Review dispositions — all three observations, at `ea4d636e`

**1. `overview-navigation.ts:33` — `buildOverviewCommunityLinks` does not exist. Accepted, corrected.**

`grep -rn buildOverviewCommunityLinks src/` returns nothing; the file exposes `overviewNodeHref`, `buildOverviewTopNodes` and `buildOverviewSearchIndex`, and its two `localeCompare` sites are the top-node tie-break at :51 and the search-index sort at :67. The comment now names those two sorts. Inline thread replied and resolved.

**2. Probe child process has no timeout. Accepted, with a justified bound.**

`runUnderLocale` now passes `timeout: 60_000`. The bound is reasoned rather than arbitrary: the slowest probe builds a git repository and a graph inside the child, ~2s locally and longer on a cold Windows runner, so 60s sits far above the real cost and far below vitest's per-test timeout. A hung child now fails as *that* probe instead of stalling the file with no indication of which one stopped.

**3. Vacuous-pass guard — accepted. Softening the collator assertion — declined, with evidence.**

*Accepted half.* The helper destructured child stdout, so two probes that died silently compared `''` to `''` and passed while proving nothing. `runUnderLocale` now parses instead: the output must be exactly `<digest> <locale>`, the digest must match `/^[0-9a-f]{64}$/`, and the collator must be non-empty. This runs on **every platform, Windows included** — an empty digest cannot pass anywhere. Verified by mutation, not assumed: making a probe print `''` fails the file, and making it print `notadigest <locale>` also fails.

*Declined half.* The non-Windows different-collator assertion stays an assertion, not a `console.warn`. It is this control's reason to exist: if both arms resolve to the same collator, the byte comparison is between two identical configurations and the control has silently stopped testing what it claims. Windows genuinely cannot satisfy it — ICU there follows the system locale and ignores `LC_ALL` — which is why Windows relies on the hermetic leg that pins the order to a pure function of the strings. Everywhere else a collapsed pair is a real loss of coverage and should fail loudly.

The suggestion rested on a small-icu or restricted-`NODE_ICU_DATA` build collapsing both arms. That is not reproducible here — `process.versions.icu = 78.2`, `icu_small = false`, and even `node --icu-data-dir=/nonexistent` still resolves `sv-SE` — and more decisively, **all six protected lanes (ubuntu, macOS, Windows × Node 20/22) pass with the assertion in place**, so it is compatible with the supported matrix. If a supported build is ever shown to collapse the pair, that reproduction would change the answer; absent one, weakening the assertion would trade a real guarantee for a hypothetical host.

No production ordering logic changed in this commit.

**Comment by mohanagy** [comment]
### Follow-up on observation 2 — the timeout bound was unsound, now corrected

The independent exact-head review caught that my 60s child timeout sat **above** vitest's per-test timeout, which `vitest.config.ts` sets to 15s (30s on Windows). The per-test timeout therefore always fired first: the `execFileSync` option was dead code and the attribution I claimed for it never happened. That was my error, not CodeRabbit's — the original suggestion was simply to add a timeout, and I chose a value that could not work.

Fixed in `14ea5945`. The bound is now two thirds of the per-test budget under the same platform rule — **10s, or 20s on Windows** — which keeps about a tenfold margin over the measured cost while still firing before vitest gives up:

```text
heaviest probe (builds a git repo + graph)   ~1s per arm, 1936ms for the two-arm test
other cross-locale probes                    72-143ms per test
```

Two guards now hold the hierarchy, because a mirrored constant nobody checks is exactly how the 60s version went unnoticed:

- one asserts the child bound is strictly below the per-test budget;
- the other reads `vitest.config.ts` and fails if `DEFAULT_TEST_TIMEOUT` moves without this file.

Both verified by mutation rather than assumed — restoring 60s fails the file, and desynchronising the mirror fails it too.

Controls now 47/47. No production code changed.

**Comment by mohanagy** [comment]
### CI note — one lane flake, re-run, unrelated to this PR

`validate (ubuntu-latest, Node 20)` failed once at head `14ea5945`:

```text
FAIL tests/unit/stdio-server.test.ts > stdio runtime
     > keeps MCP control requests responsive while another process holds the refresh lease
Error: Test timed out in 10000ms.
```

Checked before re-running rather than assumed:

- `git diff b892eafc..HEAD --name-only | grep stdio-server` → **0**. This PR does not touch that file.
- `grep -cE 'indexing-outcomes|shared/git|diff-overlay|proof-report|localeCompare|compareUnicodeCodePoints' tests/unit/stdio-server.test.ts` → **0**. It exercises the MCP stdio runtime and a refresh lease; nothing this PR changed is on that path.
- It is a responsiveness test with its own hardcoded 10s budget, on a lane that spent 610.96s in tests under 4 workers.
- The same head passed the other five lanes, **including both Windows lanes**, which are the slowest.

That is a load-sensitive pre-existing flake, so the lane was re-run rather than the test altered. Final gate status will be reported once it settles.

**Comment by mohanagy** [comment]
### HOLD-715 at `14ea5945` — two findings, both reproduced

The exact-head independent review (thread `01a03cd7`) returned HOLD. Both findings verified before accepting them. **Not merging.**

---

#### 1. The closed-domain equivalence claim is false — `az-AZ` breaks it

I claimed the three closed-domain sort domains were comparator-invariant, on an exhaustive pairwise sweep across en-US, sv-SE, da-DK, de-DE, tr-TR and mt-MT: 5,202 ordered pairs, zero disagreements. That sweep was wrong by omission.

```text
empty_extraction  vs  extractor_error      (both valid INDEXING_REASON_CODES members)

code-point   -1        en-US -1   sv -1   da -1   de -1   tr -1   mt -1
az-AZ        +1   <-- reversed
```

Azerbaijani places `x` between `h` and `ı`, so `extractor_error` sorts before `empty_extraction`. Turkish, which I did test, has no `x` in its alphabet at all — which is precisely why the sweep missed it.

End-to-end consequence, measured:

```text
LC_ALL=en_US.UTF-8   fixed ee9be46a…   mutated ee9be46a…   identical
LC_ALL=az_AZ.UTF-8   fixed ee9be46a…   mutated 03d6d603…   BYTES DIFFER
```

So mutation **M11 can change persisted bytes** on a supported host. Per the standing instruction, that means HOLD rather than carrying 10/12 forward.

**The shipped code is not wrong.** Production uses `compareUnicodeCodePoints` and emits `ee9be46a…` on both hosts — host-independent, as intended. What is wrong is my *classification*: the A2 tier claimed "no discriminating fixture exists", and one does. Those three sites have been carrying static-audit-only coverage that they did not need to.

That makes this finding an upgrade, not just a defect: with `az-AZ` as the second arm, the closed-domain sites can take real behavioural controls, M11 becomes catchable, and the only genuine survivor left is M9 — which remains uncatchable because its two inputs are equal strings by construction.

#### 2. The probe timeout hierarchy is still unsound for sequential probes

Each cross-locale test runs **two** probes in sequence, each bounded at two thirds of the per-test budget. A first probe taking over 5s (10s on Windows) pushes the second probe's deadline past the per-test budget, so vitest fires first again and attribution is lost. Both hierarchy guards pass, so they do not detect this.

The bound must satisfy `nProbes × PROBE ≤ PER_TEST`, i.e. one third of the budget — 5s, or 10s on Windows — which still leaves roughly fivefold headroom over the heaviest measured probe (~1s).

---

Neither finding is a defect in the ordering fix itself. Both are defects in my control and classification work, and both need the contract's A2 tier and the control set revised rather than patched at the example. Returning to the maintainer for direction rather than proceeding to the merge gate.

**Comment by mohanagy** [comment]
### A2 classification rewritten — the closed domains are now derived, not asserted

The HOLD was correct, and the fix is not a patch to the reported pair. The claim that the
closed domains have no discriminating behavioural fixture is **withdrawn and replaced by a
search** over every locale this Node build collates:

```text
INDEXING_REASON_CODES        406 pairs   discriminable in az, haw, lt, lv
EXTRACTION_STRATEGIES         10 pairs   no discriminator in any of 94 locales
EXTRACTION_FALLBACK_REASONS    0 pairs   one member, so structurally impossible
```

That classification is now itself a test. If a future reason code, strategy or fallback
introduces a discriminating pair into a domain recorded as having none, it fails and says to
add a behavioural control rather than widen the expectation.

**`reason_buckets` gains a real behavioural control** (`C4-reason-codes`) under en-US/az-AZ,
proving all three required properties: the pair is valid in the real domain and survives
`parseIndexingManifest`; code point and az-AZ genuinely disagree
(`empty_extraction, extractor_error` against `extractor_error, empty_extraction`); and
production emits code-point order with identical bytes across both arms.

**The mutation now fails its exact named control:**

```text
C4-reason-codes -- closed reason-code ordering
  > writes the same manifest bytes under en-US and az-AZ
  AssertionError: bytes differ between host collations: ddb05288… vs 7154a8b0…
```

Locale arms are now per fixture rather than global, because en-US agrees with code point on
the ASCII reason pair while sv-SE is what splits the accented paths. Leg (a) accordingly
requires the arms to disagree with each other and at least one to disagree with code point.

**Timeout bound solved, not chosen.** From the invariant

```text
MAX_SEQUENTIAL_PROBES_PER_TEST * PROBE_TIMEOUT_MS + SETUP_TEARDOWN_MARGIN_MS <= TEST_TIMEOUT_MS
```

with a negative control that rejects *both* bounds this file previously shipped — the flat
60s and the two-thirds value — plus the boundary case, and a per-test probe counter so the
probe count is observed rather than assumed.

**Mutation inventory re-derived — 18 mutants, 17 caught, 1 survivor:**

```text
9 production ordering sites   all caught    7 behavioural + audit,
                                            2 audit-only (the provably non-discriminable domains)
3 comparator integrity        all caught
6 control integrity           5 caught, 1 survivor
```

The single survivor is the literal-capability fixture mutation. It touches no production
file, so it cannot move persisted bytes, and the two routes digest identically because the
parser returns the capability strings it was handed — an equality now asserted inside the
control itself, so it stops being true loudly if the parser ever transforms capabilities.

Performance re-measured, because a source comment changed and the file is therefore no
longer byte-identical: **−31.09%** p50 (p10 −32.25, p90 −29.34), artifact size unchanged at
316659 bytes.

File: tests/unit/persisted-ordering-locale-determinism.test.ts [hunk 2]
```diff
@@ -0,0 +1,983 @@
...
+
+/** Every locale this build collates, so the search is over the real space. */
+const SUPPORTED_LOCALES = Intl.Collator.supportedLocalesOf([
+  'af', 'am', 'ar', 'az', 'be', 'bg', 'bn', 'bs', 'ca', 'cs', 'cy', 'da', 'de', 'el', 'en', 'eo',
+  'es', 'et', 'eu', 'fa', 'fi', 'fil', 'fo', 'fr', 'ga', 'gl', 'gu', 'ha', 'haw', 'he', 'hi', 'hr',
+  'hu', 'hy', 'id', 'ig', 'is', 'it', 'ja', 'ka', 'kk', 'kl', 'km', 'kn', 'ko', 'kok', 'ky', 'lkt',
+  'ln', 'lo', 'lt', 'lv', 'mk', 'ml', 'mn', 'mr', 'ms', 'mt', 'my', 'nb', 'ne', 'nl', 'nn', 'om',
+  'or', 'pa', 'pl', 'ps', 'pt', 'ro', 'ru', 'se', 'si', 'sk', 'sl', 'smn', 'sq', 'sr', 'sv', 'sw',
+  'ta', 'te', 'th', 'ti', 'to', 'tr', 'uk', 'ur', 'uz', 'vi', 'wae', 'wo', 'yi', 'yo', 'zh', 'zu',
+])
```
[the remark this answers is not in the document]
Comment (mohanagy) [reply tests/unit/persisted-ordering-locale-determinism.test.ts]: **Accepted and corrected** in `5f94435d`.

You are right that `Intl.Collator.supportedLocalesOf([...])` only filters the list it is handed — it does not enumerate what the build supports. Describing the sweep as covering "every locale this build collates" was false, and that is exactly the failure mode that produced the original defect: a hand-picked list silently bounded the search and reported "no discriminator" for a domain that has several.

Rather than only reword it, the candidates are now generated: every two-letter language subtag (all 676), plus a documented set of longer tags whose collations reorder Latin letters. `supportedLocalesOf` then narrows that to what this build actually collates — 116 locales here, against the previous 94 — and it finds two discriminating locales the old list missed:

```text
before   az, haw, lt, lv
after    az, ee, fy, haw, lt, lv
```

The classification is unchanged (reason codes discriminable; strategies and fallback reasons not), so the controls are unaffected.

The wording no longer claims exhaustiveness anywhere, in this file or in the audit — there is no API that enumerates every locale ICU can collate, so the search space is stated and left for the reader to judge. The anti-vacuity guard also now asserts the generation really produced its 676 subtags, so a generator that silently yielded nothing cannot make the search report "no discriminator" forever.

**Comment by mohanagy** [comment]
### CI note — Windows Node 20 lane flake, re-run

`validate (windows-latest, Node 20)` failed at head `5f94435d` with three `tests/unit/watch.test.ts` failures, all filesystem-watcher timeouts against that file's own 5000ms budget:

```text
× detects and repairs graph/source-manifest policy disagreement   10832ms
× reconciles at MCP startup and refreshes a later agent edit        9999ms
× stops cleanly when a later Git visibility snapshot fails          5025ms
Error: Timed out after 5000ms waiting for graph refresh
```

Checked more carefully than the earlier `stdio-server` flake, because `watch.ts` genuinely does read `indexingManifest.outcomes` — the collection this PR reorders:

- `git diff b892eafc..HEAD --name-only | grep watch` → **0**. Neither `watch.ts` nor `watch.test.ts` is touched.
- `watch.ts` consumes outcomes **order-independently**: `:304` builds a `Set` of paths, `:334` iterates all of them. No `[0]`, `.at(0)` or `slice` anywhere in either file.
- `watch.test.ts` is in this PR's affected-suite set and passes locally — it is one of the ten files in the 244-test run.
- The other Windows lane (Node 22) passed on this same head, as did both Ubuntu and both macOS lanes.

So the ordering change cannot reach these assertions, and the failures are watcher-timing under load. Lane re-run rather than test altered.

**Comment by mohanagy** [comment]
### Merge gate — awaiting renewed exact-head authorization

```text
head          5f94435deffcda059a16f5ebb4feef4aa4f8d90b
tree          bb782d9f158b0deb88b1430ec62e82826732ca32
base          next @ b892eafc916c8dfaddff13e585c901ebee187afe   (re-fetched, unmoved)
merge result  a8653acb4d1ad61e17e5a49a12c4121a9b7536fb           (tested)
FINAL         GO-715-ORDERING   thread 01a03d2d-8077-7922-8f57-6ed6a0b3be60
```

Protected CI, workflow run `32944573281`, every job bound to that head:

```text
validate (macos-latest,   Node 20)  success  98108814437
validate (macos-latest,   Node 22)  success  98108806868
validate (ubuntu-latest,  Node 20)  success  98108794901
validate (ubuntu-latest,  Node 22)  success  98108770809
validate (windows-latest, Node 20)  success  98108770108
validate (windows-latest, Node 22)  success  98108771605
CodeRabbit                          pass
```

Review threads: **2 of 2 resolved, 0 unresolved.** All three original observations plus the
follow-up on the search space are dispositioned on the PR.

### Mutation inventory — 18 mutants, 17 caught, 1 survivor

```text
9 production ordering sites   all caught   7 behavioural + audit
                                           2 audit-only: extraction_strategy_buckets,
                                             fallback_reason_buckets — the domains a search
                                             proves non-discriminable
3 comparator integrity        all caught   pinned locale, shadowed binding, dropped annotation
6 control integrity           5 caught     weakened path fixture, weakened reason pair,
                                             broken timeout invariant, understated probe
                                             count, misclassified domain
                              1 survivor   C2-literal-capability
```

`reason_buckets` — previously recorded as an unkillable survivor — is now killed
behaviourally by its own named control:

```text
C4-reason-codes > writes the same manifest bytes under en-US and az-AZ
  bytes differ between host collations: ddb05288… vs 7154a8b0…
```

### The one survivor

**C2-literal-capability**, the capability fixture rebuilt from literals instead of
`parseIndexingManifest` output.

1. It edits a fixture inside the control file; no production file is touched, so no
   persisted byte can move and no value can be selected differently.
2. It is unobservable in principle, not merely uncaught: the parser returns the same
   `capability` strings it was handed, so the two routes are equal by construction and
   digest identically. No assertion can separate identical inputs.
3. The equality is asserted inside the control, so if the parser ever transforms
   capabilities it fails loudly and the mutation becomes observable.
4. The route is protected structurally — route and ordering assertions live in one test.

### Timeout proof

```text
MAX_SEQUENTIAL_PROBES_PER_TEST * PROBE_TIMEOUT_MS + SETUP_TEARDOWN_MARGIN_MS <= TEST_TIMEOUT_MS
2 * 6000 + 3000 <= 15000        (Windows: 2 * 13500 + 3000 <= 30000)
```

`hierarchyHolds` is pure, so the negative control can exercise it: it rejects both bounds
this file previously shipped — the flat 60s and the two-thirds value that one probe survived
and two did not — plus the boundary `PROBE_TIMEOUT_MS + 1`. A per-test probe counter makes
the probe count observed rather than assumed. Heaviest measured probe: ~1s.

### Standard gates

```text
persisted-ordering controls   57 / 57
ten affected suites          244 / 244
typecheck                    clean
build                        success
qualify:validate             contract v1.0.0 consistent
release:verify               validated
pack:dry-run                 436 files, 941.4 kB
worktree after build/pack    clean, local == remote
```

### Carried forward

`git diff cc037bb5..HEAD -- src/` is empty, so production is byte-identical to the head at
which these were measured:

- performance **−31.09%** p50 (p10 −32.25, p90 −29.34);
- artifact size 316659 bytes, identical on both locale arms;
- base reproduction at `b892eafc`; scope adjudication; #716/#717/#718/#719 classifications.

### Lifecycle

PR body says **`Implements #715.`** — no `Closes`/`Fixes`/`Resolves` in title or body,
scanned. #715 stays open until the squash merge is followed by successful push-triggered
protected CI on `next`. #716 remains release-blocking. No release work and no #659 work
started.

**Not merged.** Awaiting renewed authorization for this exact head; if it moves, the
authorization is void.

**Comment by mohanagy** [comment]
### Independent final review — exact head

```text
verdict     GO-715-ORDERING
threadId    01a03d2d-8077-7922-8f57-6ed6a0b3be60
head        5f94435deffcda059a16f5ebb4feef4aa4f8d90b
tree        bb782d9f158b0deb88b1430ec62e82826732ca32
base        next @ b892eafc916c8dfaddff13e585c901ebee187afe
```

The thread is distinct from the design and prequalification threads and from all seven
retired historical threads. It reviewed that exact head and tree. The reviewer ran read-only
and made **0** production modifications.

**Comment by mohanagy** [comment]
Maintainer exact-head merge authorization:
yes

Authorized head:
5f94435deffcda059a16f5ebb4feef4aa4f8d90b

Authorized tree:
bb782d9f158b0deb88b1430ec62e82826732ca32

Authorized tested merge result:
a8653acb4d1ad61e17e5a49a12c4121a9b7536fb

Independent verdict:
GO-715-ORDERING

Protected workflow:
32944573281 — six of six successful

#715:
remains open pending post-merge push-CI verification

#716:
next release-blocking issue after #715

#659:
not started

Release:
none

**Comment by mohanagy** [comment]
### Merged — squash at the authorized exact head

```text
squash commit   7d65ea1ae7b51243bfb367e98cbb9742b8721ad3
squash tree     bb782d9f158b0deb88b1430ec62e82826732ca32   (identical to the PR tree)
parent          b892eafc916c8dfaddff13e585c901ebee187afe   (the authorized base)
target branch   next
merged at       2026-08-26T09:28:47Z
```

The merge was performed with `--match-head-commit 5f94435deffcda059a16f5ebb4feef4aa4f8d90b`,
without `--admin`, and with an explicit subject and body rather than the concatenated commit
log. `git diff 5f94435d 7d65ea1a` is empty, so the merged content is byte-identical to the
qualified head.

Squash message:

```text
fix(indexing): make persisted collection ordering host-independent

Implements #715.

Qualified at exact head 5f94435deffcda059a16f5ebb4feef4aa4f8d90b with independent
verdict GO-715-ORDERING.
```

Unchanged by this merge:

```text
#715 #716 #717 #718 #719 #659   all still open
main                            3371ada8425efa7f8cabdac781fa227feaea7a6a (2026-08-12)
npm @lubab/madar latest         0.32.1
latest release                  v0.32.1 (2026-08-11)
tags on the squash commit       0
```

#715 stays open until push-triggered protected CI on `next` at `7d65ea1a` is verified.

**Comment by mohanagy** [comment]
### Declared controlled rerun of one failed lane

Recording the first attempt in full **before** rerunning, so the rerun cannot be mistaken for
the original result.

```text
push workflow   32953215772
squash commit   7d65ea1ae7b51243bfb367e98cbb9742b8721ad3
attempt 1       5 lanes passed, 1 lane failed
failed lane     validate (windows-latest, Node 20)   job 98129116720
failure         tests/unit/workspace.test.ts:258
                  linked worktree artifact routing >
                  routes conventional out paths outside the linked source checkout
```

**In that failing lane**, all twelve #715 suites executed **and passed**. The lane's only
failing test file was `workspace.test.ts`:

```text
Test Files  1 failed | 315 passed (316)
     Tests  1 failed | 5267 passed | 1 expected fail | 37 skipped (5306)

FAIL lines in the entire lane: 1   (tests/unit/workspace.test.ts)

ten load-bearing suites            0 FAIL lines each — all passed
  indexing-completeness  indexing-generate  spi-diff-overlay  proof-report
  export  wiki  telemetry  doctor  watch  compare
two #715 control suites            0 FAIL lines each — all passed
  persisted-ordering-locale-determinism   persisted-ordering-callsite-policy
```

**Execution-integrity counters in that failing lane were all zero:**

```text
worker-start failures                 0
worker-response / handshake failures  0
silently omitted modules              0
absorbed infrastructure failures      0
```

**The merged #715 tree does not modify the failing test or its production import closure.**
The twelve changed files do not include `tests/unit/workspace.test.ts`,
`src/shared/security.ts`, or `tests/unit/helpers/phase-run.ts`. `src/shared/security.ts` and
`src/shared/workspace.ts` each have a transitive relative-import closure of size 1 — they
import no local module — so the failing assertion's code path cannot reach any #715 change.

**The identical tree passed the same lane in the PR workflow.** Tree
`bb782d9f158b0deb88b1430ec62e82826732ca32`, `validate (windows-latest, Node 20)`, job
`98108770108`, PR run `32944573281`: success.

**Windows Node 22 passed the same test at the same squash commit.** Job `98129116635` in this
same run executed `routes conventional out paths outside the linked source checkout` at
`7d65ea1a` and passed, 316/316.

### What this rerun is

A **declared controlled attempt** to distinguish a transient environment / path-resolution
condition from a reproducible defect. Only the failed job is rerun; the five successful lanes
are not. No tracked file changes between the two attempts, so the two attempts differ only in
runner environment.

Attempt 1 is **not** superseded and will not be described as green. Both attempts will be
reported. If the failure reproduces, it is treated as a defect to investigate, not as a
transient — and #715 stays open either way until the result is interpreted.

**Comment by mohanagy** [comment]
### Post-merge verification complete — two attempts, both reported

```text
push workflow   32953215772   event=push   branch=next
squash commit   7d65ea1ae7b51243bfb367e98cbb9742b8721ad3
final attempt   2             run conclusion: success
```

**Attempt 1 — failed.** `validate (windows-latest, Node 20)`, job `98129116720`:
`tests/unit/workspace.test.ts:258`, *routes conventional out paths outside the linked source
checkout*. One failing test file out of 316; the other five lanes passed.

**Controlled attempt 2 — passed, with no repository change.** Same lane, job `98198745591`,
same squash commit, same tree. `git` was untouched between attempts: no commit, no rebase, no
force-push, no tracked-file edit. Only the failed job was rerun — the five successful lanes
retain their attempt-1 start timestamps (09:28Z) against the rerun's 13:38:26Z, so they were
carried forward, not re-executed.

```text
                              attempt 1              controlled attempt 2
job id                        98129116720            98198745591
run attempt                   1                      2
checked-out SHA               7d65ea1a               7d65ea1a
conclusion                    failure                success
test files                    315 passed, 1 failed   316 passed, 0 failed
tests                         5267 passed, 1 failed  5268 passed, 0 failed
workspace.test.ts             FAILED at :258         PASSED (1368ms)
"escapes the allowed dir"     1 occurrence           0 occurrences
FAIL lines in lane            1                      0
worker-start failures         0                      0
worker-response/handshake     0                      0
omitted modules               0                      0
absorbed infra failures       0                      0
ten load-bearing suites       all passed             all passed
two #715 control suites       all passed             all passed
```

Note that in **attempt 1 as well**, every one of the ten load-bearing suites and both #715
control suites passed. The single failure was in a suite #715 does not touch.

### Classification

**Non-reproduced Windows environment / path-resolution failure.**

Not called a known flake. The `escapes the allowed directory` signature appears in none of the
previous thirteen failed CI runs, so no recurring family is established — one occurrence and
one non-reproduction is exactly that, and nothing more. The realpath / 8.3-short-name /
junction mechanism remains an unproven hypothesis. If it recurs, that evidence would establish
a family; this record does not.

Attempt 1 is **not** superseded and is not described as green.

### Final six-lane state

All six jobs successful, all six independently verified from their raw logs:

```text
lane                                job          concl.   sha        files  ws wr om ab suites ctrls
validate (macos-latest,   Node 20)  98198746432  success  7d65ea1a   316/0   0  0  0  0  10/10  2/2
validate (macos-latest,   Node 22)  98198774563  success  7d65ea1a   316/0   0  0  0  0  10/10  2/2
validate (ubuntu-latest,  Node 20)  98198745767  success  7d65ea1a   316/0   0  0  0  0  10/10  2/2
validate (ubuntu-latest,  Node 22)  98198747245  success  7d65ea1a   316/0   0  0  0  0  10/10  2/2
validate (windows-latest, Node 20)  98198745591  success  7d65ea1a   316/0   0  0  0  0  10/10  2/2
validate (windows-latest, Node 22)  98198747178  success  7d65ea1a   316/0   0  0  0  0  10/10  2/2

execution integrity clean 6/6      lane conclusions green 6/6
```

- all six lanes checked out `7d65ea1a`;
- 316 test files executed per lane;
- all ten #715 load-bearing suites passed in every lane;
- both #715 control suites passed in every lane;
- zero worker-start signatures;
- zero worker-response / handshake signatures;
- zero omitted modules;
- zero absorbed infrastructure failures.

### Merge integrity

```text
squash commit   7d65ea1ae7b51243bfb367e98cbb9742b8721ad3
squash tree     bb782d9f158b0deb88b1430ec62e82826732ca32   identical to the qualified tree
parent          b892eafc916c8dfaddff13e585c901ebee187afe   the authorized base
git diff 5f94435d 7d65ea1a → empty
```

Post-merge verification is satisfied. #715 is complete.
